google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/FieldValue.java

```diff
@@ -183,7 +184,7 @@ public long getTimestampValue() {
     // timestamps are encoded in the format 1408452095.22 where the integer part is seconds since
     // epoch (e.g. 1408452095.22 == 2014-08-19 07:41:35.220 -05:00)
     BigDecimal secondsWithMicro = new BigDecimal(getStringValue());
-    BigDecimal scaled = secondsWithMicro.scaleByPowerOfTen(6);
+    BigDecimal scaled = secondsWithMicro.scaleByPowerOfTen(6).setScale(0, RoundingMode.HALF_UP);
```

Comment:
Let's add a comment here as to why rounding is needed.

Reply:
Done. 
I will leave the issue open.